src/libcore/iter/iterator.rs

```diff
@@ -258,6 +258,32 @@ pub trait Iterator {
         None
     }
 
+    /// Returns an iterator adaptor that goes over the original iterator,
+    /// skipping `step` elements at a time.
+    ///
+    /// **Panics** if `step == 0`.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// let a = [0, 1, 2, 3, 4, 5];
+    ///
+    /// let mut iter = a.iter().every_nth(2);
+    ///
+    /// assert_eq!(iter.next(), Some(&0));
+    /// assert_eq!(iter.next(), Some(&2));
+    /// assert_eq!(iter.next(), Some(&4));
+    /// assert_eq!(iter.next(), None);
+    /// ```
+    #[inline]
+    #[stable(feature = "rust1", since = "1.0.0")]
```

Comment:
Yes, I plan to do that. I just didn't know how when I made the pull request.  Also, the feature name depends on whether we get rid of the `Step` trait or if we'll keep the step trait and have this as `every_nth`. I personally prefer the former, since there's some code that already uses `step_by` and wouldn't break as bad with a change in the implementation. That would mean, however, that I'd have to reimplement `Iterator for Range`, which is not hard, but would definitely take some time. Once all that gets figured out, I'll put the proper attributes.